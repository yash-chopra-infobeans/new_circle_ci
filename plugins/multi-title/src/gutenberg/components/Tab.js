import Input from './WrappedInput';

const { Slot } = wp.components;
const { forwardRef } = wp.element;

export const Tab = forwardRef((props, ref) => {
  const { prefixRef, titleRef } = ref;
  const {
    tab,
    getCurrentTitle,
    getCurrentPlaceholder,
    getCurrentPrefix,
    handleTitleChange,
    getCharLimitValue,
    getCharLimit,
    handleAdditionalFieldsChange,
    getAdditionalFieldValue,
    onTitleBlur,
    onPrefixBlur,
  } = props;
  const currentTitle = getCurrentTitle(tab);
  const placeholder = getCurrentPlaceholder(tab);
  const prefixTitle = getCurrentPrefix(tab);

  // remove className so that of tab properties can be passed to WrappedInput
  const {
    additionalFields,
    className,
    combineCharLimit,
    charLimit,
    prefix: prefixProperties,
    ...tabProperties
  } = tab;

  let additionalFieldsRender = additionalFields?.render || false;

  if (!additionalFieldsRender && typeof additionalFields === 'function') {
    additionalFieldsRender = additionalFields;
  }

  return (
    <div className="editor-post-title tabbed-title display">
      <div className="wp-block editor-post-title__block">
        {tab.prefix && (
          <Input
            name={prefixProperties.name || tabProperties.name}
            titleType="prefix"
            title={prefixProperties.title}
            placeholder="Kicker"
            value={prefixTitle}
            className={`title-PrefixInput title-PrefixInput-${tab.name}`}
            onChange={handleTitleChange(tab, 'prefix')}
            onBlur={onPrefixBlur(tab, 'prefix')}
            showCharLimitAt={0}
            showCharLimitOnFocus
            charLimitValue={getCharLimitValue(tab, 'value')}
            charLimit={getCharLimit(tab, 'prefix')}
            ref={prefixRef}
          />
        )}
        <Input
          name={tabProperties.name}
          titleType="value"
          title={tabProperties.title}
          placeholder={placeholder}
          value={currentTitle}
          className={`title-input title-input-${tab.name} title-input--main`}
          onChange={handleTitleChange(tab, 'value')}
          onBlur={onTitleBlur(tab, 'value')}
          showCharLimitAt={0}
          showCharLimitOnFocus
          charLimitValue={getCharLimitValue(tab, 'prefix')}
          charLimit={getCharLimit(tab, 'value')}
          ref={titleRef}
        />
        {additionalFieldsRender
          ? additionalFieldsRender({
              ...props,
              tab,
              handleChange: handleAdditionalFieldsChange,
              getValue: getAdditionalFieldValue,
            })
          : null}
        <Slot name={`multi-title-tab-${tab.name}`} />
      </div>
    </div>
  );
});

export default Tab;
