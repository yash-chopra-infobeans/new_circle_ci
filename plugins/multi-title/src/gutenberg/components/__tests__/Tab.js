import { shallow } from 'enzyme';
import { Tab } from '../Tab';

const { createRef } = wp.element;
const refs = {
  prefixRef: createRef(),
  titleRef: createRef(),
};

export const exampleTab = {
  name: 'headline',
  title: 'Headline 2',
  className: 'tab tab-headline',
  metaKey: 'multi_title_headline',
  combineCharLimit: true,
  prefix: {
    enabled: true,
    metaKey: 'multi_title_headline_prefix',
  },
};

describe('<Tab />', () => {
  it('should render without additionalFields', () => {
    const componentProps = {
      attributes: {
        titles: {
          headline: {
            value: 'test headline',
            prefix: 'test prefix',
          },
        },
      },
      tabs: [exampleTab],
      tab: exampleTab,
      activeTab: 'headline',
      getCurrentTitle: jest.fn(),
      getCurrentPlaceholder: jest.fn(),
      getCurrentPrefix: jest.fn(),
      handleTitleChange: jest.fn(),
      getCharLimitValue: jest.fn(),
      getCharLimit: jest.fn(),
      handleAdditionalFieldsChange: jest.fn(),
      getAdditionalFieldValue: jest.fn(),
      onPrefixBlur: jest.fn(),
      onTitleBlur: jest.fn(),
    };
    const component = shallow(<Tab {...componentProps} ref={refs}/>);

    expect(component).toMatchSnapshot();
  });

  it('should render additionalFields when passed as function', () => {
    exampleTab.additionalFields = () => <p>test additional field render</p>;

    const componentProps = {
      attributes: {
        titles: {
          headline: {
            value: 'test headline',
            prefix: 'test prefix',
          },
        },
      },
      tabs: [exampleTab],
      tab: exampleTab,
      activeTab: 'headline',
      getCurrentTitle: jest.fn(),
      getCurrentPlaceholder: jest.fn(),
      getCurrentPrefix: jest.fn(),
      handleTitleChange: jest.fn(),
      getCharLimitValue: jest.fn(),
      getCharLimit: jest.fn(),
      handleAdditionalFieldsChange: jest.fn(),
      getAdditionalFieldValue: jest.fn(),
      onPrefixBlur: jest.fn(),
      onTitleBlur: jest.fn(),
    };
    const component = shallow(<Tab {...componentProps} ref={refs}/>);

    expect(component).toMatchSnapshot();
    expect(component.find('p').text()).toEqual('test additional field render');
  });

  it('should render additionalFields when passed as object', () => {
    exampleTab.additionalFields = {
      render: () => <p>test additional field render</p>,
    };

    const componentProps = {
      attributes: {
        titles: {
          headline: {
            value: 'test headline',
            prefix: 'test prefix',
          },
        },
      },
      tabs: [exampleTab],
      tab: exampleTab,
      activeTab: 'headline',
      getCurrentTitle: jest.fn(),
      getCurrentPlaceholder: jest.fn(),
      getCurrentPrefix: jest.fn(),
      handleTitleChange: jest.fn(),
      getCharLimitValue: jest.fn(),
      getCharLimit: jest.fn(),
      handleAdditionalFieldsChange: jest.fn(),
      getAdditionalFieldValue: jest.fn(),
      onPrefixBlur: jest.fn(),
      onTitleBlur: jest.fn(),
    };
    const component = shallow(<Tab {...componentProps} ref={refs}/>);

    expect(component).toMatchSnapshot();
    expect(component.find('p').text()).toEqual('test additional field render');
  });
});
