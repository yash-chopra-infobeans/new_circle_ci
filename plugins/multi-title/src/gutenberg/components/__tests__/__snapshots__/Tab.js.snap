// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> should render additionalFields when passed as function 1`] = `
<div
  className="editor-post-title tabbed-title display"
>
  <div
    className="wp-block editor-post-title__block"
  >
    <ForwardRef(withForwardedRef(WrappedInput))
      className="title-PrefixInput title-PrefixInput-headline"
      name="headline"
      placeholder="Kicker"
      showCharLimitAt={0}
      showCharLimitOnFocus={true}
      titleType="prefix"
    />
    <ForwardRef(withForwardedRef(WrappedInput))
      className="title-input title-input-headline title-input--main"
      name="headline"
      showCharLimitAt={0}
      showCharLimitOnFocus={true}
      title="Headline 2"
      titleType="value"
    />
    <p>
      test additional field render
    </p>
    <Slot
      name="multi-title-tab-headline"
    />
  </div>
</div>
`;

exports[`<Tab /> should render additionalFields when passed as object 1`] = `
<div
  className="editor-post-title tabbed-title display"
>
  <div
    className="wp-block editor-post-title__block"
  >
    <ForwardRef(withForwardedRef(WrappedInput))
      className="title-PrefixInput title-PrefixInput-headline"
      name="headline"
      placeholder="Kicker"
      showCharLimitAt={0}
      showCharLimitOnFocus={true}
      titleType="prefix"
    />
    <ForwardRef(withForwardedRef(WrappedInput))
      className="title-input title-input-headline title-input--main"
      name="headline"
      showCharLimitAt={0}
      showCharLimitOnFocus={true}
      title="Headline 2"
      titleType="value"
    />
    <p>
      test additional field render
    </p>
    <Slot
      name="multi-title-tab-headline"
    />
  </div>
</div>
`;

exports[`<Tab /> should render without additionalFields 1`] = `
<div
  className="editor-post-title tabbed-title display"
>
  <div
    className="wp-block editor-post-title__block"
  >
    <ForwardRef(withForwardedRef(WrappedInput))
      className="title-PrefixInput title-PrefixInput-headline"
      name="headline"
      placeholder="Kicker"
      showCharLimitAt={0}
      showCharLimitOnFocus={true}
      titleType="prefix"
    />
    <ForwardRef(withForwardedRef(WrappedInput))
      className="title-input title-input-headline title-input--main"
      name="headline"
      showCharLimitAt={0}
      showCharLimitOnFocus={true}
      title="Headline 2"
      titleType="value"
    />
    <Slot
      name="multi-title-tab-headline"
    />
  </div>
</div>
`;
